lib/ansible/module_utils/urls.py

```diff
@@ -302,6 +302,16 @@ def match_hostname(cert, hostname):
 
     HAS_MATCH_HOSTNAME = True
 
+if hasattr(socket, 'create_connection'):
+    create_connection = socket.create_connection
+else:
+    # Just a graceful fallback for py < 2.6, implementing old behaviour.
+    # Wouldn't work with IPv6, but it is unlikely that python 2.4 could be met
+    # on IPv6-only machine.
+    def create_connection(address):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect(address)
+        return sock
```

Comment:
I'm not really a fan of this approach, and generally speaking, we have tried to stay away from it.

I am more in favor of just doing:

```
try:
    sock = socket.create_connection(address)
except AttributeError:
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.connect(address)
```